go_modules/lib/dependabot/go_modules/file_parser.rb

```diff
@@ -78,17 +69,18 @@ def module_info
               end
 
               File.write("go.mod", go_mod_content)
+              File.write("main.go", "package dummypkg\n func main() {}\n")
 
-              command = "go mod edit -print > /dev/null"
-              command += " && go list -m -json all"
+              command = "go get > /dev/null" # Verify the packages are installable
```

Comment:
This does add a bit of overhead, and as discussed in the PR body, I doubt that it's necessary. The upside is that we fail early in the sad path, the downside is we have to duplicate error handling between this class and the `GoModUpdater`, and it adds some overhead in the happy path.

Reply:
Would this slow down the pull request opening latency? I assume beause we're running this twice, this means there will be significant overhead as you meant. I'm in favor of removing it. In fact maye this method's only concern should be `return the list of required packages`. If those packages are installable or valid should be handled probably somewhere else. Thoughts on that ?

Reply:
> Would this slow down the pull request opening latency?

A bit, it's hard to say how significant it would be, we definitely perform some heavier operations in the actual updating step, but I noticed our tests for this this class increased by ~40 seconds after adding this `go get` vs `main`, and our test `go.mod` files are typically very small. When we remove the `go get`, we speed up this step significantly as we do not need to make any network calls.

> I'm in favor of removing it. In fact maye this method's only concern should be return the list of required packages. If those packages are installable or valid should be handled probably somewhere else. Thoughts on that ?

I tend to agree! 👍

Reply:
Yeah 👍  on removing `go get` from this step, seems like we're overloading the parser by doing resolvability that should be handled by the update checker/file updater.